### PR TITLE
test(sandbox): share offline search contracts across providers

### LINKED ISSUE #5675: [test] Share offline sandbox search contracts across providers
### Before you start

- [x] I searched [existing issues](https://github.com/bytedance/deer-flow/issues?q=is%3Aissue) and this is not a duplicate.

### Problem / motivation

Sandbox search behavior is covered by shared command/parser tests and separate provider suites, but common scenarios are maintained independently. This makes it easy for a provider to miss a case that another provider already handles.

I propose a small offline test-only follow-up: run the same stable `ls`, `glob` and `grep` cases through Local, AIO, E2B, BoxLite, Tenki and OpenSandbox. This is not a replacement for the existing suites or a new production abstraction.

### Proposed solution

Use one parameterized scenario suite and small execution/RPC fixtures. Shell-based adapters run their production commands through a real local POSIX shell against temporary files. AIO's file RPC fixture independently reads that tree. Tests do not substitute precomputed final provider answers, provision sandboxes, or call models.

Initial cases cover visible and ignored descendants, spaces/quotes/Unicode paths, directory globbing, grep line numbers and case sensitivity, single-file grep, root-relative grep filters, empty/no-match results, missing roots, invalid regex, genuine overflow, transport failures, and missing search programs. Compare semantic results while allowing existing root-entry/directory-suffix differences.

Keep the first slice independent of ongoing fixes: add ignored-root cases after #5667, literal metacharacters after #5627, and exact-cap local cases after #5491. Existing provider-specific tests remain authoritative for their SDK and lifecycle behavior. No permanent xfails are proposed.

### Affected area(s)

Sandbox / Docker, Docs

### Alternatives considered

Continuing to copy fixtures and assertions into each provider's test file works locally but does not ensure that all providers receive each shared scenario. Replacing production provider interfaces or adding a live-cloud matrix would be unnecessary for this first slice.

### Additional context

The prepared implementation adds one test module plus test guidance and has 112 passing offline cases. POSIX execution is required for this tier; it does not prove live SDK/server compatibility, provisioning, authentication, or lifecycle correctness. No production code is changed. The opt-in live model smoke work in #4754 and per-user path isolation in #3087 have different scopes.

### PR DESCRIPTION
**Pull request by Totoro-qaq**
Fixes #5675

## Why

Common sandbox search cases are currently maintained in separate provider suites. This adds one offline scenario suite that exercises the same `ls`, `glob` and `grep` behavior through all six current adapters, reducing the chance that a new shared boundary case reaches only one implementation's tests.

## What changed

Add 112 parameterized cases with a shared temporary filesystem and small execution/RPC doubles. Shell adapters execute their production commands using local sh/find/grep; AIO's file RPC double reads the fixture tree independently. Tests cover results, ignored descendants, special paths, grep scope/line numbers/case, empty and missing paths, regex errors, true overflow, transport failures and unavailable binaries. Document provider presentation differences and deferred cases in test guidance.

No production code changes. Existing SDK and lifecycle suites remain. Known pending fixes (#5667, #5627, #5491) retain their own owners; their additional shared scenarios can follow after merge.

## Surface area

- [ ] Frontend UI
- [ ] Backend API
- [ ] Agents / LangGraph
- [ ] Sandbox runtime changes
- [ ] Skills
- [ ] Dependencies
- [ ] Default behavior change
- [x] Docs / tests / CI only

## Validation

- New shared suite: 112 passed, no skips on the local POSIX host.
- `make lint`: passed.
- `make test-blocking-io`: 149 passed.
- `make test`: 18,137 passed / 15 failed / 208 skipped / 3 deselected. Clean baseline: 18,025 passed with the identical 15 failed node IDs; no new failing test IDs. The shared failures are browser/fetch URL-validation cases in this environment, including private/loopback-address rejections.
- Agent guidance and `git diff --check`: passed.

This tests adapter behavior offline, not live SDK/server compatibility or provisioning. The POSIX execution tier is explicitly skipped on Windows; it neither makes cloud/model calls nor replaces provider-specific tests.

## AI assistance

**Tool(s) used:** Codex.

**How you used it:** Audited existing test seams and active ownership, implemented the shared scenarios and fixtures, and ran the checks above. Prepared as a draft for human review.

- [x] I've read and understand every line of this change and take responsibility for it — it's not unreviewed AI output.

### CODE REVIEW

File: backend/tests/test_sandbox_search_contract.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+from pathlib import Path
+from types import SimpleNamespace
+
+import pytest
+
+from deerflow.sandbox.local.local_sandbox import LocalSandbox
+
+PROVIDERS = ("local", "aio", "e2b", "boxlite", "tenki", "opensandbox")
+REMOTE_PROVIDERS = PROVIDERS[1:]
+SHELL_SEARCH_PROVIDERS = PROVIDERS[2:]
```
Comment (willem-bd) [inline backend/tests/test_sandbox_search_contract.py]: Nit: `REMOTE_PROVIDERS` and `SHELL_SEARCH_PROVIDERS` are derived by position, so provider membership is coupled to the tuple order in `PROVIDERS`. Since this suite is explicitly meant to be extended (AGENTS.md: "Add those shared scenarios after their fixes land"), consider deriving the groups explicitly, e.g. `REMOTE_PROVIDERS = tuple(p for p in PROVIDERS if p != "local")` plus a named `SHELL_SEARCH_PROVIDERS`, or at least a short comment noting the positional coupling — inserting a future RPC-based provider mid-tuple currently only surfaces as a confusing failure in `test_missing_search_program_does_not_become_no_matches`.

**Review by willem-bd — COMMENTED** [review]
Reviewed against head `4b2dba19`. Verified the suite passes (112/112) against this head's production code on a second POSIX host (macOS), confirmed the execution/RPC doubles faithfully mirror the production seams (AIO `grep_files`/`find_files`/`list_path`/`exec_command` kwargs including `max_file_size="1M"`, E2B `commands.run`, BoxLite/Tenki `_sh`, OpenSandbox `_run` with `logs.stdout[].text`/`exit_code`/`error`), and that all six adapters validate regex client-side so the `re.error` assertion holds everywhere. `ruff format --check` and `ruff check` are clean under `backend/ruff.toml`, and the suite coexists cleanly with `test_remote_list_dir.py`. The deferral of #5667/#5627/#5491 with explicit anti-xfail guidance in the test AGENTS.md is a nice touch. No correctness or convention issues found; one nit below.

File: backend/tests/test_sandbox_search_contract.py (hunk 1 above)
Comment (Totoro-qaq) [reply backend/tests/test_sandbox_search_contract.py]: Addressed in effe1740: remote providers are now selected by name, and shell-search providers are listed explicitly, removing the positional dependency. Shared contract suite plus `test_remote_list_dir.py`: 137 passed; lint/format checks passed.

**Review by WillemJiang — APPROVED** [review]
